content/posts/custom-rake-tasks.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "Custom Rake Tasks"
+preview: "How to add custom Rake tasks"
+status: "Learn how to add custom Rake tasks to your Hanami project"
+date: 2018-11-20T14:49:44+01:00
+draft: false
+gems: ["hanami"]
+tags: ["hanami", "rake"]
+versions: ["1.0"]
+---
+
+If you want to add customer Rake tasks to your Hanami project, you have two ways: `Rakefile` and `.rake` files.
+
+## Rakefile
+
+`Rakefile` is the main Rake file of the project.
+Add here your Rake tasks to perform general related activities.
+
+The special `:environment` task can be used as a task dependency to load the entire code of the project.
+
+```ruby
+# Rakefile
+namespace :db do
+  desc "Database backup"
+  task backup: :environment do
+    # ...
+  end
+end
+```
+
+## `.rake` files
+
+If your list of Rake tasks is growing, you may want to split them per categories in several files:
+
+```shell
+$ mkdir -p rakelib
+```
+
+```ruby
+# rakelib/catalog.rake
+namespace :catalog do
```

Comment:
also, I found that you need to load `config/boot.rb` in some cases (when you work with dry-system for example. WDYT to adding this point too?

Reply:
Do you have an example?

Reply:
will make it in few seconds

Reply:
## without booting
![screenshot 2018-11-20 18 06 35](https://user-images.githubusercontent.com/1147484/48782381-2496c680-ecef-11e8-8d7f-4a688662e8a7.jpeg)
![screenshot 2018-11-20 18 06 39](https://user-images.githubusercontent.com/1147484/48782382-252f5d00-ecef-11e8-91f1-8b9fb147a411.jpeg)

## with booting
![screenshot 2018-11-20 18 06 58](https://user-images.githubusercontent.com/1147484/48782403-2d879800-ecef-11e8-9b92-dd26ff82b2e1.jpeg)
![screenshot 2018-11-20 18 07 02](https://user-images.githubusercontent.com/1147484/48782405-2e202e80-ecef-11e8-8e9b-65bc96e4af6d.jpeg)

Reply:
@davydovanton Adding `:environment` as dependency of the task doesn't do the trick?

```ruby
task tasks: :environment do
  # ...
end
```

Can you please tell me if it works? Thanks!

Reply:
checked, all works correct 👍

Reply:
Thanks!